javascript/javascript3/week3/index.js

```diff
@@ -0,0 +1,186 @@
+let screenshotUrl;
+const userEmail = document.querySelector(".userEmail");
+const userPassword = document.querySelector(".userPassword");
+const form = document.querySelector(".mainForm");
+const options = {
+  method: "GET",
+  headers: {
+    "X-RapidAPI-Key": `${RAPID_API_KEY}`,
+    "X-RapidAPI-Host": `${RAPID_API_HOST}`,
+  },
+};
+//Get an URL of website from user and pub its screenshot
+
+async function getScreenshot() {
+  const inputElement = document.querySelector(".userInput");
+  try {
+    const response = await fetch(
+      `https://${RAPID_API_HOST}/screenshot?url=https%3A%2F%2F${inputElement.value}&width=1920&height=1080&fullscreen=true`,
+      options
+    );
+    const data = await response.json();
+    screenshotUrl = data.screenshotUrl;
+    showScreenshot();
+    document.querySelector(".messageForUser").innerHTML = "";
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+function showPubToUser() {
+  document.querySelector(".messageForUser").innerHTML =
+    "Please wait... Publishing your screenshot";
+}
+function showSendingProcess() {
+  document.querySelector(".messageForUser").innerHTML =
+    "Your screenshot was published on the server";
+}
+function showScreenshot() {
+  const image = document.querySelector(".screenshot");
+  image.src = `${screenshotUrl}`;
+  image.alt = "website-screenshot";
+}
+async function postData() {
+  const inputElement = document.querySelector(".userInput");
+  const userEmailValue = userEmail.value;
+  try {
+    const userInput = inputElement.value;
+    await fetch(`https://crudcrud.com/api/${CRUDCRUD_API_KEY}/savedlinks`, {
+      method: "POST",
+      headers: {
+        "Content-type": "application/json",
+      },
+      body: JSON.stringify({ userInput, screenshotUrl, userEmailValue }),
+    });
+    showSendingProcess();
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+// Get data (list of saved screenshots and website links).
+async function getData() {
+  document.querySelector(".screenshot").classList.add("hide");
+  document.querySelector(".messageForUser").innerHTML = "";
+  try {
```

Comment:
General comment here.

When using a try - catch block, it's really good practice to handle as little code as possible in your try block, so that you can handle it more specifically in your catch block. When the errors get more complicated, being able to handle them individually rather than in one big catch helps us build more stable and debuggable code 👍